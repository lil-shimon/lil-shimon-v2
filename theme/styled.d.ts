import "styled-components";

declare module "styled-components" {
  export interface DefaultTheme {
    mediaQuery: {
      sp: string;
      tablet: string;
      laptop: string;
    };
  }
}
