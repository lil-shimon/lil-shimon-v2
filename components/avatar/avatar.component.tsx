import Image from "next/image";
import * as Styles from "./avatar.styles";

export const Avatar = () => {
  return (
    <Styles.Container>
      <Image src="/images/shimosawa.png" alt="shimosawa profile image" fill />
    </Styles.Container>
  );
};
