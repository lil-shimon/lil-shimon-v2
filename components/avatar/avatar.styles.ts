import styled from "styled-components";

export const Container = styled.div`
  position: relative;
  border-radius: 50%;
  height: 140px;
  width: 140px;
  overflow: hidden;

  img {
    object-fit: cover;
  }
`;
