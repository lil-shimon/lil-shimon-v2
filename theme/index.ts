import { DefaultTheme } from "styled-components";

const customMediaQuery = (minWidth: number) =>
  `@media (min-width: ${minWidth}px)`;

export const theme: DefaultTheme = {
  mediaQuery: {
    sp: customMediaQuery(576),
    tablet: customMediaQuery(768),
    laptop: customMediaQuery(992),
  },
};
