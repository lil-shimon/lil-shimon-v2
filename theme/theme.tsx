import { ReactNode } from "react";
import { ThemeProvider as StyledComponentThemeProvider } from "styled-components";

import { DefaultTheme } from "styled-components";

const customMediaQuery = (minWidth: number) =>
  `@media (min-width: ${minWidth}px)`;

const theme: DefaultTheme = {
  mediaQuery: {
    sp: customMediaQuery(576),
    tablet: customMediaQuery(768),
    laptop: customMediaQuery(992),
  },
};

type Props = {
  children: ReactNode;
};

export const ThemeProvider = ({ children }: Props) => {
  return (
    <StyledComponentThemeProvider theme={theme}>
      {children}
    </StyledComponentThemeProvider>
  );
};
