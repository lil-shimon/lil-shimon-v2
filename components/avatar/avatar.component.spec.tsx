import { render } from "@testing-library/react";
import { Avatar } from "./avatar.component";
import { expect } from "vitest";
import { ThemeProvider } from "@/theme";

describe("Avatar", () => {
  it("should render", () => {
    const { asFragment } = render(<Avatar />, {
      wrapper: ({ children }) => <ThemeProvider>{children}</ThemeProvider>,
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
