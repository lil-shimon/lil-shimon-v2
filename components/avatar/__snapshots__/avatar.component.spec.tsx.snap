// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should render 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb jRwtSY"
  >
    <img
      alt="shimosawa profile image"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fshimosawa.png&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Fshimosawa.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fshimosawa.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fshimosawa.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fshimosawa.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fshimosawa.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fshimosawa.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fshimosawa.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fshimosawa.png&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
</DocumentFragment>
`;
